### PR TITLE
Add yamllint to infrastructure

### PR DESCRIPTION
**Pull request by catalinailie**
Trello card: https://trello.com/c/oUuP3Crp/2872-govuk-puppet-hieradataaws-integrationyaml-file-2

### CODE REVIEW

File: modules/yamllint/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Installs yamlliv, a linter for YAML files: https://yamllint.readthedocs.io/en/stable/
+#
+# === Parameters:
+#
+# [*version*]
+#   Version to install. Defaults to '1.28.0'
+#
+class yamllint(
+  $version = '1.28.0',
```
Comment (ChrisBAshton) [inline modules/yamllint/manifests/init.pp:11]: Hmm this differs from the `1.26.3` version on Debian you've linked to above 🤔

File: modules/govuk/manifests/node/s_apt.pp [hunk 2]
```diff
@@ -109,6 +109,11 @@
       location => 'http://download.draios.com/stable/deb',
       release  => 'stable-amd64/',
       key      => 'D27A72F32D867DF9300A241574490FD6EC51E8C4';
+    'yamllint':
+      location => 'https://debian.pkgs.org/sid/debian-main-amd64/yamllint_1.26.3-1_all.deb.html',
+      release  => 'stable',
+      repos    => ['main'],
+      key      => '72ECF46A56B4AD39C907BBB71646B01B86E50310';
```
Comment (ChrisBAshton) [inline modules/govuk/manifests/node/s_apt.pp:116]: Can you explain where this key came from?

**Review by ChrisBAshton — COMMENTED** [review]
Thanks for persevering with what's been a harder card than expected!
A couple of points below, and also a request to make the commit message more descriptive (link to Trello, explain why this code is being added, etc).

**Comment by catalinailie** [comment]
Since puppet is going away, getting this installed is not a priority. The only purpose for this package was to check if there are any duplicate keys in the hieradata aws config.

**Pull request closed without merging**